### PR TITLE
Python 3.6 baggage code removal

### PR DESCRIPTION
**Pull request by hallvictoria**
## Description

Fixes # https://github.com/Azure/azure-functions-pyfx-planning/issues/139 // https://github.com/Azure/azure-functions-pyfx-planning/issues/176

---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: python/prodV2/worker.config.json
Comment (vrdmr) [inline python/prodV2/worker.config.json:1]: We can remove ProdV2 and ProdV3 itself.

File: pack/Microsoft.Azure.Functions.V3.PythonWorker.nuspec
Comment (vrdmr) [inline pack/Microsoft.Azure.Functions.V3.PythonWorker.nuspec:1]: Do we need non-V4 nuspecs?

File: tests/unittests/test_utilities_dependency.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def test_initialize_in_linux_consumption(self):
         sys.path.extend([
             '/tmp/functions\\standby\\wwwroot',
             '/home/site/wwwroot/.python_packages/lib/site-packages',
-            '/azure-functions-host/workers/python/3.6/LINUX/X64',
```
Comment (vrdmr) [inline tests/unittests/test_utilities_dependency.py:60]: Might as well move to a newer version?

File: setup.cfg [hunk 2]
```diff
@@ -1,23 +0,0 @@
-[tool:pytest]
```
Comment (vrdmr) [inline setup.cfg:1]: Aren't we using this?

**Comment by vrdmr** [comment]
We should be able to remove aio_compat.py as well as rethink if we need typing_inspect.py itself.

File: python/prodV2/worker.config.json
Comment (hallvictoria) [reply python/prodV2/worker.config.json:1]: Done

File: pack/Microsoft.Azure.Functions.V3.PythonWorker.nuspec
Comment (hallvictoria) [reply pack/Microsoft.Azure.Functions.V3.PythonWorker.nuspec:1]: Nope, removed those since we removed V2 & V3

File: tests/unittests/test_utilities_dependency.py (hunk 1 above)
Comment (hallvictoria) [reply tests/unittests/test_utilities_dependency.py:60]: Updated all to 3.11

File: setup.cfg (hunk 2 above)
Comment (hallvictoria) [reply setup.cfg]: I think it's outdated. We can remove both this and setup.py whenever / if we move to pyproject.toml

**Review by vrdmr — APPROVED** [review]
LGMT :shipit: 

Take a look at the docker tests as well to ensure it passes.

File: tests/unittests/setup.cfg [hunk 3]
```diff
@@ -13,11 +13,8 @@ warn_return_any = True
 disallow_subclassing_any = False
 ignore_missing_imports = True
 
-[mypy-azure_functions_worker._thirdparty.aio_compat]
-ignore_errors = True
```
Comment (gavin-aguiar) [inline tests/unittests/setup.cfg:17]: why is this moved to unittest dir?

Comment (hallvictoria) [reply tests/unittests/setup.cfg:17]: fixed

**Review by gavin-aguiar — APPROVED** [review]
